vault/resource_oci_auth_backend.go

```diff
@@ -0,0 +1,250 @@
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		CustomizeDiff: getMountCustomizeDiffFunc(consts.FieldPath),
+		Schema: map[string]*schema.Schema{
+			consts.FieldPath: {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Unique name of the auth backend to configure.",
+				Default:     consts.MountTypeOCI,
+				// standardize on no beginning or trailing slashes
+				StateFunc: func(v interface{}) string {
+					return strings.Trim(v.(string), "/")
+				},
+			},
+			consts.FieldDescription: {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			consts.FieldAccessor: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The accessor of the auth backend",
+			},
+			consts.FieldOCIHomeTenancyID: {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: " The Tenancy OCID of your OCI account.",
+				Sensitive:   true,
+			},
+			consts.FieldTune: authMountTuneSchema(),
+		},
+	}, false)
+
+	// Add common automated root rotation schema to the resource.
+	provider.MustAddSchema(r, provider.GetAutomatedRootRotationSchema())
+
+	return r
+}
+
+func ociAuthBackendWrite(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client, e := provider.GetClient(d, meta)
+	if e != nil {
+		return diag.FromErr(e)
+	}
+
+	path := d.Get(consts.FieldPath).(string)
+	desc := d.Get(consts.FieldDescription).(string)
+
+	log.Printf("[DEBUG] Enabling %s auth backend %q", consts.AuthMethodOCI, path)
+	err := client.Sys().EnableAuthWithOptionsWithContext(ctx, path, &api.EnableAuthOptions{
+		Type:        consts.AuthMethodOCI,
+		Description: desc,
+	})
+
+	if err != nil {
+		return diag.Errorf("error enabling %s auth backend %q: %s", consts.AuthMethodOCI, path, err)
+	}
+	log.Printf("[DEBUG] Enabled %s auth backend %q", consts.AuthMethodOCI, path)
+
+	d.SetId(path)
+
+	return ociAuthBackendUpdate(ctx, d, meta)
+}
+
+func ociAuthBackendUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client, e := provider.GetClient(d, meta)
+	if e != nil {
+		return diag.FromErr(e)
+	}
+
+	ociPath := d.Id()
+	ociAuthPath := "auth/" + ociPath
+	path := ociAuthBackendConfigPath(ociPath)
+
+	if !d.IsNewResource() {
+		newMount, err := util.Remount(d, client, consts.FieldPath, true)
+		if err != nil {
+			return diag.FromErr(err)
+		}
+
+		ociAuthPath = "auth/" + newMount
+		path = ociAuthBackendConfigPath(newMount)
+
+		if d.HasChanges(consts.FieldIdentityTokenKey, consts.FieldDescription) {
+			desc := d.Get(consts.FieldDescription).(string)
+			config := api.MountConfigInput{
+				Description: &desc,
+			}
+			if err := client.Sys().TuneMountWithContext(ctx, path, config); err != nil {
+				return diag.FromErr(err)
+			}
+		}
+	}
+
+	data := map[string]interface{}{}
+
+	if d.HasChange(consts.FieldOCIHomeTenancyID) {
+		data[consts.FieldOCIHomeTenancyID] = d.Get(consts.FieldOCIHomeTenancyID)
+	}
```

Comment:
```suggestion
	data[consts.FieldOCIHomeTenancyID] = d.Get(consts.FieldOCIHomeTenancyID)
```

This actually must be checked every time regardless of a change, because it is a required field.